### ISSUE TITLE
fluentd 0.14.6 and elastic plugin 1.7.0 error_class=NoMethodError error="undefined method `msgpack_each'

### ISSUE THREAD

**Issue by Kuckkuck** [uken/fluent-plugin-elasticsearch#200]
I get the following error with fluentd 0.14.7 and elasticsearch plugin 1.7.0. Last working version is fluentd 0.14.4 and elasticsearch plugin 1.7.0. Any idea?

  2016-10-11 09:52:49 +0000 [warn]: suppressed same stacktrace
2016-10-11 09:52:58 +0000 [warn]: [Fluent::ElasticsearchOutput] failed to flush the buffer. plugin_id="object:1946b40" retry_time=5 next_retry=2016-10-11 09:53:09 +0000 chunk="53e93d5ce80f898894f40d4f0293eb23" error_class=NoMethodError error="undefined method `msgpack_each' for #<Fluent::Plugin::Buffer::MemoryChunk:0x000000029f8858>"
  2016-10-11 09:52:58 +0000 [warn]: suppressed same stacktrace
2016-10-11 09:53:09 +0000 [warn]: [Fluent::ElasticsearchOutput] failed to flush the buffer. plugin_id="object:1946b40" retry_time=6 next_retry=2016-10-11 09:53:18 +0000 chunk="53e93d5ce80f898894f40d4f0293eb23" error_class=NoMethodError error="undefined method `msgpack_each' for #<Fluent::Plugin::Buffer::MemoryChunk:0x000000029f8858>"

**Comment by jimmycuadra** [comment]
Same problem here. Here's fluentd's log, easier to read than OP's:

```
2016-10-15 00:52:43 +0000 [warn]: failed to flush the buffer. plugin_id="object:2ad39f93a718" retry_time=0 next_retry=2016-10-15 00:52:44 +0000 chunk="53edcc1369922316389cfc5252627b50" error_class=NoMethodError error="undefined method `msgpack_each' for #<Fluent::Plugin::Buffer::MemoryChunk:0x0055a74184a538>"
  2016-10-15 00:52:45 +0000 [warn]: /usr/local/bundle/gems/fluent-plugin-elasticsearch-1.7.0/lib/fluent/plugin/out_elasticsearch.rb:263:in `write'
  2016-10-15 00:52:45 +0000 [warn]: /usr/local/bundle/gems/fluentd-0.14.7/lib/fluent/compat/output.rb:127:in `write'
  2016-10-15 00:52:45 +0000 [warn]: /usr/local/bundle/gems/fluentd-0.14.7/lib/fluent/plugin/output.rb:995:in `try_flush'
  2016-10-15 00:52:45 +0000 [warn]: /usr/local/bundle/gems/fluentd-0.14.7/lib/fluent/plugin/output.rb:1188:in `flush_thread_run'
  2016-10-15 00:52:45 +0000 [warn]: /usr/local/bundle/gems/fluentd-0.14.7/lib/fluent/plugin/output.rb:393:in `block (2 levels) in start'
  2016-10-15 00:52:45 +0000 [warn]: /usr/local/bundle/gems/fluentd-0.14.7/lib/fluent/plugin_helper/thread.rb:66:in `block in thread_create'
```

Finalized fluent.conf:

```
<source>
  @type systemd
  pos_file "/var/log/fluent/systemd.pos"
  tag "journald"
  read_from_head true
</source>
<match journald>
  @type rewrite_tag_filter
  rewriterule1 CONTAINER_NAME ^k8s_([^\.]+) kubernetes.container.$1
  rewriterule2 CONTAINER_NAME (.+) docker.container.$1
</match>
<filter kubernetes.**>
  @type kubernetes_metadata
  use_journal true
</filter>
<match kubernetes.container.fluentd>
  @type null
</match>
<match **>
  @type aws-elasticsearch-service
  logstash_format true
  include_tag_key true
  time_key "time"
  reload_connections false
  <endpoint>
    url "https://REDACTED"
    region "REDACTED"
  </endpoint>
  <buffer>
    flush_mode interval
    retry_type exponential_backoff
  </buffer>
  <parse>
    time_key time
  </parse>
  <inject>
    time_key time
    tag_key tag
  </inject>
</match>
```

Installed gems:

* activesupport (5.0.0.1)
* addressable (2.4.0)
* aws-sdk (2.6.7)
* aws-sdk-core (2.6.7)
* aws-sdk-resources (2.6.7)
* bundler (1.13.3)
* concurrent-ruby (1.0.2)
* cool.io (1.4.5)
* domain_name (0.5.20160826)
* elasticsearch (1.0.18)
* elasticsearch-api (1.0.18)
* elasticsearch-transport (1.0.18)
* excon (0.53.0)
* faraday (0.9.2)
* faraday_middleware-aws-signers-v4 (0.1.1)
* ffi (1.9.14)
* fluent-mixin-config-placeholders (0.4.0)
* fluent-plugin-aws-elasticsearch-service (0.1.6)
* fluent-plugin-elasticsearch (1.7.0)
* fluent-plugin-kubernetes_metadata_filter (0.25.3)
* fluent-plugin-rewrite-tag-filter (1.5.5)
* fluent-plugin-s3 (0.7.1)
* fluent-plugin-systemd (0.0.4)
* fluentd (0.14.7)
* http (0.9.8)
* http-cookie (1.0.3)
* http-form_data (1.0.1)
* http_parser.rb (0.6.0)
* i18n (0.7.0)
* jmespath (1.3.1)
* kubeclient (1.1.4)
* lru_redux (1.1.0)
* mime-types (3.1)
* mime-types-data (3.2016.0521)
* minitest (5.9.1)
* msgpack (1.0.0)
* multi_json (1.12.1)
* multipart-post (2.0.0)
* netrc (0.11.0)
* recursive-open-struct (1.0.0)
* rest-client (2.0.0)
* serverengine (2.0.0)
* sigdump (0.2.4)
* strptime (0.1.8)
* systemd-journal (1.2.3)
* thread_safe (0.3.5)
* tzinfo (1.2.2)
* tzinfo-data (1.2016.7)
* unf (0.1.4)
* unf_ext (0.0.7.2)
* uuidtools (2.1.5)
* yajl-ruby (1.2.1)

**Comment by jimmycuadra** [comment]
This appears to have been resolved by fluent/fluentd#1273, which was released in fluentd 0.14.8.
